src/pages/organizers/[organizerId]/ownerships/index.page.tsx

```diff
@@ -55,6 +73,34 @@ const Ownership = () => {
   const approvedRequests = requestsByState[RequestState.APPROVED] ?? [];
   const pendingRequests = requestsByState[RequestState.REQUESTED] ?? [];
 
+  const approveOwnershipRequestMutation = useApproveOwnershipRequestMutation({
+    onSuccess: async () => {
+      await queryClient.invalidateQueries('ownership-requests');
+      setIsMutationSuccessful(true);
+      setIsQuestionModalVisible(false);
```

Comment:
You should also clear the other state:

```tsx
setSelectedRequest(undefined);
setActionType(undefined);
```

Also you can just check if the actionType and request is defined to show the modal

```tsx
const isQuestionModalVisible = !!actionType && !!selectedRequest;
```

Reply:
I can’t clear the selected request here because I need the email, organization name, and action type for the success alert. However, I can clear them once the user closes the alert.